### PR TITLE
test: add framework backbone and event/scheduling test coverage

### LINKED ISSUE #1167: Improve test coverage for event handling and scheduling APIs
## Description

The event handling and scheduling modules are the APIs app authors touch on every automation — `P.*` predicates, `C.*` conditions, `A.*` accessors on every `bus.on_state_change`/`bus.on` call, the `Listener` dataclass that backs every subscription, `Bus` pub/sub delivery, and the `ScheduledJob`/trigger classes behind every `scheduler.run_*` call. Coverage on these modules currently trails the rest of the codebase, per the latest `coverage.xml`:

| Module | Current coverage | Missed lines |
|---|---|---|
| `src/hassette/event_handling/predicates.py` | 44% | 150 |
| `src/hassette/event_handling/conditions.py` | 37% | 117 |
| `src/hassette/event_handling/accessors.py` | 37% | 87 |
| `src/hassette/bus/listeners.py` | 46% | 149 |
| `src/hassette/bus/bus.py` | 77% | 66 |
| `src/hassette/scheduler/classes.py` | 37% | 117 |
| `src/hassette/scheduler/triggers.py` | 59% | 60 |

746 missed lines total. This is Tier 2 of the broader test coverage initiative (project-wide ~61% → 80%+); this tier targets the user-facing event/scheduling APIs specifically, since bugs here affect every automation an app author writes.

## Acceptance Criteria

- [ ] `src/hassette/event_handling/predicates.py` reaches 85%+ line coverage (combinators `AllOf`/`AnyOf`/`Not`, per-predicate classes like `ValueIs`, `DidChange`, `StateComparison`, `AttrComparison`, `ServiceDataWhere`, and the `compare_value`/`normalize_where`/`ensure_tuple` helpers)
- [ ] `src/hassette/event_handling/conditions.py` reaches 85%+ line coverage (`Glob`, `Regex`, `Contains`/`NotIn`/`Intersects` family, `Comparison`, `Increased`/`Decreased`, and their `_format_collection`/`_validate_collection` helpers)
- [ ] `src/hassette/event_handling/accessors.py` reaches 85%+ line coverage (state/attr get_* accessor factories, `get_all_changes`/`_recursive_get_differences`, service-data accessors)
- [ ] `src/hassette/bus/listeners.py` reaches 85%+ line coverage (`ListenerIdentity`, `ListenerOptions`, `HandlerInvoker`, `DurationConfig`, `Listener`, `_duration_configs_match`, `make_async_handler`)
- [ ] `src/hassette/bus/bus.py` reaches 85%+ line coverage (`add_listener`/`_resolve_collision`/`_resolve_and_register` collision paths, `emit`, `_on_internal`/`_subscribe` dispatch)
- [ ] `src/hassette/scheduler/classes.py` reaches 85%+ line coverage (`CronTrigger` DST/ambiguous-time handling, `ScheduledJob.matches`/`diff_fields`/`mark_registered`/`cancel`)
- [ ] `src/hassette/scheduler/triggers.py` reaches 85%+ line coverage (`After`, `Once`, `Every`, `Daily`, `Cron` — `first_run_time`/`next_run_time`/`trigger_*` properties)
- [ ] No regressions in existing tests; new tests follow the patterns in `tests/TESTING.md` (unit tests use `make_mock_hassette()` / direct instantiation, integration tests use `HassetteHarness`)
- [ ] Tests pass — `uv run nox -s tests_with_coverage` run and green, coverage report confirms each module at 85%+

## Affected Areas

- `src/hassette/event_handling/predicates.py` — no source changes; add unit tests
- `src/hassette/event_handling/conditions.py` — no source changes; add unit tests
- `src/hassette/event_handling/accessors.py` — no source changes; add unit tests
- `src/hassette/bus/listeners.py` — no source changes; add unit tests
- `src/hassette/bus/bus.py` — no source changes; add unit/integration tests
- `src/hassette/scheduler/classes.py` — no source changes; add unit tests
- `src/hassette/scheduler/triggers.py` — no source changes; add unit tests
- `tests/unit/bus/test_predicates.py`, `test_conditions.py`, `test_listeners.py`, `test_listener_identity.py`, `test_listener_options.py`, `test_bus.py` — extend existing suites rather than duplicating coverage
- `tests/unit/test_triggers.py`, `tests/unit/test_scheduler_job_names.py` — extend for `classes.py`/`triggers.py` gaps
- `tests/integration/bus/*`, `tests/integration/test_scheduler.py` — extend where behavior needs real bus/scheduler wiring (see `tests/TESTING.md` decision table)

## Context

Coverage-only work — no production code should change as a byproduct unless a test surfaces an actual bug (in which case, follow the project's TDD bug workflow in `CLAUDE.md`: reproduce, write a failing test, fix, verify).

Existing test files already partially cover these modules (`tests/unit/bus/test_predicates.py`, `test_conditions.py`, `test_listeners.py`, `test_listener_identity.py`, `test_listener_options.py`, `tests/unit/test_triggers.py`) — extend them rather than creating parallel files, so patterns stay consistent and coverage tooling doesn't double-count.

Per `tests/TESTING.md`, use `HassetteHarness` for bus routing / scheduler firing / state propagation behavior that needs real component wiring, and direct unit tests (`make_mock_hassette()` or plain instantiation) for isolated dataclass/predicate/condition logic. Mock only at boundaries (HA API, time, filesystem) — see `references/common/testing.md` invariants.

`CronTrigger` in `scheduler/classes.py` has DST-sensitive branches (`_is_fall_back_ambiguous`, `_dst_safe_from_dt`) that are easy to undertest with naive datetimes — use `whenever` per the project's date/time convention when constructing fixtures.

### LINKED ISSUE #1168: Improve test coverage for framework backbone modules (Tier 1)
## Description

The framework backbone modules that everything depends on have significant coverage gaps. These are the modules where bugs cause total system failure -- if `core.py` or `websocket_service.py` break, nothing works. Currently ~654 missed lines across 7 files, with coverage ranging from 0% (`restart.py`) to 76% (`app_lifecycle_service.py`). Target: 85%+ per module.

This is Tier 1 of a broader initiative to bring overall project coverage from ~61% to 80%+.

## Current Coverage

Measured via `pytest --cov` across unit + integration suites:

| File | Stmts | Miss | Branch | Cover | Target |
|---|---|---|---|---|---|
| `core/core.py` | 448 | 208 | 120 | 54% | 85%+ |
| `core/websocket_service.py` | 350 | 135 | 84 | 63% | 85%+ |
| `core/service_watcher.py` | 258 | 103 | 64 | 62% | 85%+ |
| `core/app_lifecycle_service.py` | 288 | 95 | 80 | 70% | 85%+ |
| `resources/base.py` | 344 | 124 | 112 | 68% | 85%+ |
| `resources/service.py` | 102 | 38 | 24 | 63% | 85%+ |
| `resources/restart.py` | — | — | — | 0%* | 85%+ |

*`restart.py` has tests (`test_restart_spec.py`, `test_restart_budget.py`) but coverage reports 0% due to early import before coverage instrumentation starts. The module is a frozen dataclass with no logic, so the real gap is elsewhere.

## Acceptance Criteria

- [ ] `core/core.py` reaches 85%+ branch coverage -- specifically: `_shutdown_children` timeout/force-terminal path, `shutdown()` total-timeout branch, duplicate-child-type validation in `wire_services()`, `_on_children_stopped` ordering, `send_event` early-return when streams closed
- [ ] `core/websocket_service.py` reaches 85%+ branch coverage -- specifically: `subscribe_events` filtered path, `authenticate()` timeout expiry, `cleanup()` partial-failure ordering
- [ ] `core/service_watcher.py` reaches 85%+ branch coverage -- specifically: `log_service_event` unchanged-status skip, concurrent restart of multiple instances, `on_service_running` non-Service child path, `register_internal_event_listeners` predicate wiring
- [ ] `core/app_lifecycle_service.py` reaches 85%+ branch coverage -- specifically: `handle_change_event` reconciliation merge logic, degraded-mode paths in `reconcile_app_registrations` (3 independent except branches), `initialize_instances`/`shutdown_instance` timeout paths, `resolve_only_app` multi-only error
- [ ] `resources/base.py` reaches 85%+ branch coverage -- specifically: `_shutdown_children` with timeout, `_auto_wait_dependencies` paths, `_run_hooks` error handling, `restart()` flow, `cleanup()` timeout
- [ ] `resources/service.py` reaches 85%+ branch coverage -- specifically: `__init_subclass__` warning for missing restart_spec, `_serve_wrapper` cancellation/exception paths, `_force_terminal` status transitions
- [ ] All new tests follow project conventions: `HassetteHarness` for integration, mock boundaries only, `# boundary-exempt:` annotations where needed
- [ ] Existing tests still pass -- no regressions

### Source files (no changes expected, coverage targets only)
- `src/hassette/core/core.py` -- main coordinator, 54% coverage
- `src/hassette/core/websocket_service.py` -- WebSocket connection service, 63% coverage
- `src/hassette/core/service_watcher.py` -- service supervision/restart, 62% coverage
- `src/hassette/core/app_lifecycle_service.py` -- app lifecycle management, 70% coverage
- `src/hassette/resources/base.py` -- Resource base class, 68% coverage
- `src/hassette/resources/service.py` -- Service base class, 63% coverage
- `src/hassette/resources/restart.py` -- RestartSpec dataclass, 0% (instrumentation artifact)

### Test files (new tests or additions to existing)
- `tests/unit/core/test_hassette_lifecycle.py` -- extend for shutdown/startup edge cases
- `tests/unit/core/test_service_watcher_exhausted.py` -- extend for multi-instance, status-skip
- `tests/unit/core/test_app_lifecycle_service.py` -- extend for timeout, degraded-mode paths
- `tests/unit/core/test_app_lifecycle_service_operations.py` -- extend for handle_change_event
- `tests/unit/core/test_websocket_readiness_events.py` -- extend for auth timeout, cleanup
- `tests/unit/resources/` -- extend for base.py shutdown/restart/cleanup, service.py serve wrapper
- `tests/integration/test_core.py` -- extend for wire_services validation, shutdown ordering
- `tests/integration/test_service_watcher.py` -- extend for multi-instance restart
- `tests/integration/test_websocket_service.py` -- extend for subscribe_events, partial cleanup

### Testing patterns to follow
- **Startup races**: gate with `asyncio.Event`, signal a second event on await, wait on that signal before asserting (no `await asyncio.sleep(0)`)
- **Timeout paths**: use `anyio.fail_after` with short timeouts and slow mock coroutines
- **Error isolation**: verify exceptions inside `execute()` don't propagate to callers
- **Mock boundaries**: mock aiohttp sessions, WebSocket connections, and DB calls -- not internal methods. Use `build_fake_ws()` for WebSocket tests. Annotate with `# boundary-exempt:` when stubbing collaborators
- **`HassetteHarness`** for integration tests wiring real components; `make_mock_hassette()` for unit tests needing a hassette mock with real config

### Priority ordering
1. `core.py` (54%) and `service_watcher.py` (62%) -- lowest coverage, highest blast radius
2. `resources/base.py` (68%) and `resources/service.py` (63%) -- foundation for all components
3. `websocket_service.py` (63%) and `app_lifecycle_service.py` (70%) -- already have decent coverage, mostly need edge-case/error-path tests
4. `restart.py` -- investigate the 0% instrumentation issue; actual test gaps are minimal

### PR DESCRIPTION
**Pull request by NodeJSmith**
### Tier 1 — Framework Backbone Coverage

- Added 52 tests for `core/core.py` and `core/websocket_service.py` — the coordinator and WS connection are the modules where a bug takes down the whole system, so branch coverage there (shutdown ordering, reconnection, subscribe-filtering paths) matters more than average.
- Added 39 tests for `core/service_watcher.py` and `core/app_lifecycle_service.py` covering supervision and reconciliation, including the three independent `except` branches in degraded-mode app reconciliation.
- Added 49 tests for `resources/base.py`, `resources/service.py`, and `resources/restart.py`. `restart.py` previously reported 0% — it's a frozen dataclass with no real logic, so the gap was a coverage-instrumentation artifact (early import before tracing starts) rather than an untested module.

### Tier 2 — User-Facing API Coverage

- Added 149 tests for `event_handling/predicates.py`, `conditions.py`, and `accessors.py` — the `P.*`/`C.*`/`A.*` surface every app author touches on `bus.on_state_change`/`bus.on` calls.
- Added ~30 tests for `bus/listeners.py` and `bus/bus.py` covering the `Listener` dataclass internals and collision/dispatch paths behind every subscription.
- Added 100 tests for `scheduler/classes.py` and `scheduler/triggers.py`, including `CronTrigger` DST/ambiguous-time handling.

### Coverage Configuration

- Excluded codegen and pure-data modules (`models/states/**`, `models/entities/**`, `types/enums.py`, `types/types.py`, `const/**`, `web/models.py`, `schemas/**`) from both `codecov.yml` and the local `pyproject.toml` `coverage.run.omit` list — these are generated or declaration-only files with no branch logic, so counting them against coverage was diluting the signal on modules that actually need tests.
- Fixed the `TYPE_CHECKING` exclusion regex (`if (typing\.)?TYPE_CHECKING:`) so `if typing.TYPE_CHECKING:` blocks are excluded the same way bare `if TYPE_CHECKING:` blocks already were.
- Reported coverage moves 55% → 67%; real coverage is higher than that because `conftest` imports modules before `pytest-cov` tracing starts, undercounting import-time statements.

### Housekeeping

- Replaced a broad `warnings.catch_warnings` suppression in the `wired_hassette` fixture teardown with a file-level `pytestmark` scoped to anyio's `ResourceWarning`. The underlying streams are already closed explicitly in teardown — the warning fires later because pytest holds a fixture-result reference that delays GC finalization, not because of an actual leak.

---

Closes #1167, closes #1168

Related to #1165, #1166 — Tier 3 and Tier 4 are separate follow-ups.

### CODE REVIEW

File: tests/unit/scheduler/test_cron_trigger_internals.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+        A per-second cron ("* * * * * *", 6-field) anchored 3 hours (10800 ticks) before
+        current_time exceeds MAX_CRON_ITERATIONS (10000), forcing the skip-ahead branch.
+        """
+        anchor = zdt(2025, 8, 18, 0, 0, 0)
+        current_time = anchor.add(hours=3)
+        assert MAX_CRON_ITERATIONS < 3 * 3600, "test setup must exceed MAX_CRON_ITERATIONS"
+
+        trigger = CronTrigger("* * * * * *", start=anchor)
+        result = trigger.first_run_time(current_time)
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/unit/scheduler/test_cron_trigger_internals.py]: Fixed — patched MAX_CRON_ITERATIONS to 5 instead of spinning through 10k iterations.

File: tests/unit/scheduler/test_cron_trigger_internals.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+    def test_exceeding_max_iterations_result_is_timezone_correct(self) -> None:
+        """The skip-ahead result still carries the trigger's original timezone."""
+        anchor = zdt(2025, 8, 18, 0, 0, 0)
+        current_time = anchor.add(hours=3)
+
+        trigger = CronTrigger("* * * * * *", start=anchor)
+        result = trigger.first_run_time(current_time)
+
+        assert result.tz == TZ
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/unit/scheduler/test_cron_trigger_internals.py:113]: Fixed — same patch applied to this test.

File: tests/unit/core/test_websocket_service_coverage.py [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+        self, websocket_service: WebsocketService
+    ) -> None:
+        """cleanup() resolves every pending response future with RetryableConnectionClosedError."""
+        fut = websocket_service.hassette.loop.create_future()
+        websocket_service._response_futures[7] = fut
+        websocket_service._ws = None
+        websocket_service._session = None
+        websocket_service._recv_task = None
+
+        with patch.object(type(websocket_service).__bases__[0], "cleanup", new=AsyncMock()):
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/unit/core/test_websocket_service_coverage.py]: Fixed in a prior commit — replaced `type(ws).__bases__[0]` with `patch.object(Service, "cleanup", ...)`.

File: tests/unit/core/test_websocket_service_coverage.py [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+        """cleanup() calls send_json(unsubscribe_events) once per active subscription, then clears the set."""
+        websocket_service._ws = build_fake_ws()
+        websocket_service._subscription_ids = {1, 2}
+        websocket_service._session = None
+        websocket_service._recv_task = None
+
+        send_json_mock = AsyncMock()
+        websocket_service.send_json = send_json_mock  # collaborator of cleanup
+
+        with patch.object(type(websocket_service).__bases__[0], "cleanup", new=AsyncMock()):
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/unit/core/test_websocket_service_coverage.py]: Same fix applied — all 4 occurrences now use `Service` directly.

File: tests/unit/core/test_websocket_service_coverage.py [hunk 5]
```diff
@@ -0,0 +1,394 @@
...
+        """cleanup() skips the unsubscribe loop (and leaves subscription_ids untouched) when ws.closed is True."""
+        websocket_service._ws = build_fake_ws(is_closed=True)
+        websocket_service._subscription_ids = {1}
+        websocket_service._session = None
+        websocket_service._recv_task = None
+
+        send_json_mock = AsyncMock()
+        websocket_service.send_json = send_json_mock  # collaborator of cleanup
+
+        with patch.object(type(websocket_service).__bases__[0], "cleanup", new=AsyncMock()):
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/unit/core/test_websocket_service_coverage.py]: Fixed in prior commit — all uses now patch `Service` directly.

File: tests/unit/core/test_websocket_service_coverage.py [hunk 6]
```diff
@@ -0,0 +1,394 @@
...
+        websocket_service._subscription_ids = set()
+
+        recv_task = asyncio.create_task(asyncio.sleep(100))
+        websocket_service._recv_task = recv_task
+
+        fake_session = MagicMock()
+        fake_session.close = AsyncMock()
+        websocket_service._session = fake_session
+
+        with patch.object(type(websocket_service).__bases__[0], "cleanup", new=AsyncMock()):
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/unit/core/test_websocket_service_coverage.py]: Fixed — same change applied to all 4 occurrences.

File: tests/unit/bus/test_handler_invoker.py [hunk 7]
```diff
@@ -277,3 +280,68 @@ async def handler(extra: str = "") -> None:
...
+            kwargs=None,
+            options=ListenerOptions(mode="single"),
+        )
+
+        async def slow_invoke() -> None:
+            await asyncio.sleep(0.05)
+
+        await invoker.dispatch(slow_invoke)
+
+        assert warn_calls == [0.01], "warn_stalled should fire once, with the patched threshold"
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/unit/bus/test_handler_invoker.py:323]: Increased the sleep margin from 0.05s to 0.2s (20x the 0.01s threshold) to reduce CI flakiness risk.

File: tests/unit/core/test_app_lifecycle_service_coverage.py [hunk 8]
```diff
@@ -0,0 +1,358 @@
...
+    async def test_stop_failure_prevents_start_and_does_not_raise(self, lifecycle_service: AppLifecycleService) -> None:
+        """If stop_app raises, reload_app catches it and never calls start_app."""
+        # boundary-exempt: collaborator of reload_app
+        lifecycle_service.stop_app = AsyncMock(side_effect=RuntimeError("stop blew up"))
+        lifecycle_service.start_app = AsyncMock()  # boundary-exempt: collaborator of reload_app
+
+        await lifecycle_service.reload_app("test_app")
+
+        lifecycle_service.start_app.assert_not_called()
+
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/unit/core/test_app_lifecycle_service_coverage.py:94]: These are annotated with `# boundary-exempt: collaborator of <method>` per the project's mocking convention. The SUT's own method is stubbed to isolate the branch under test, not to avoid calling it — the annotation makes this traceable.

**Comment by NodeJSmith** [comment]
Addressed the review findings from the latest CodeRabbit pass:

- **`warn_stalled` annotations** — added `# boundary-exempt: collaborator of dispatch` to both monkeypatch sites in `test_handler_invoker.py`
- **`wait_ready`/`restart` annotations** — added `# branch-isolation:` to the guard and error-forcing stubs in `test_service_watcher_coverage.py`
- **`svc_a.restart`/`svc_b.restart` annotations** — added `# boundary-exempt: collaborator of restart_service`
- **Warning assertion** — `test_restarts_all_matching_services_and_warns` now asserts `watcher.logger.warning.assert_called_once()`
- **Watchdog timing** — acknowledged as a duplicate; the 0.2s margin (20x threshold) is the pragmatic tradeoff. Mocking `call_later` would require patching the event loop scheduler, which is disproportionate for this test.

File: tests/unit/core/test_websocket_service_coverage.py [hunk 9]
```diff
@@ -0,0 +1,399 @@
...
+
+        async def flaky_connect_ws(_session):
+            nonlocal attempts
+            attempts += 1
+            if attempts < 3:
+                raise RuntimeError("transient connect failure")
+
+        websocket_service.connect_ws = flaky_connect_ws  # collaborator of make_connection
+        websocket_service.partial_cleanup = AsyncMock()  # collaborator of make_connection
+        websocket_service.start_recv_and_subscribe = AsyncMock(return_value="recv-task-sentinel")
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/unit/core/test_websocket_service_coverage.py]: Fixed — all stub comments in this file now use the canonical `# boundary-exempt: collaborator of <method>` prefix per tests/TESTING.md.

File: tests/unit/core/test_websocket_service_coverage.py [hunk 10]
```diff
@@ -0,0 +1,399 @@
...
+        """make_connection propagates InvalidAuthError immediately without retrying (NON_RETRYABLE)."""
+        attempts = 0
+
+        async def bad_auth_connect_ws(_session):
+            nonlocal attempts
+            attempts += 1
+            raise InvalidAuthError("token rejected")
+
+        websocket_service.connect_ws = bad_auth_connect_ws  # collaborator of make_connection
+        websocket_service.partial_cleanup = AsyncMock()  # collaborator of make_connection
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/unit/core/test_websocket_service_coverage.py]: Fixed — same canonical prefix applied.

File: tests/unit/core/test_websocket_service_coverage.py [hunk 11]
```diff
@@ -0,0 +1,399 @@
...
+            await websocket_service.raw_recv()
+
+    async def test_raw_recv_ignores_binary_frame(self, websocket_service: WebsocketService) -> None:
+        """raw_recv logs and returns without dispatching for a BINARY frame."""
+        fake_ws = build_fake_ws()
+        fake_ws.receive = AsyncMock(return_value=SimpleNamespace(type=WSMsgType.BINARY, data=b"\x00\x01"))
+        websocket_service._ws = fake_ws
+
+        dispatch_mock = AsyncMock()
+        websocket_service.dispatch = dispatch_mock  # collaborator of raw_recv
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/unit/core/test_websocket_service_coverage.py]: Fixed — updated to `# boundary-exempt: collaborator of raw_recv`.

File: tests/unit/core/test_websocket_service_coverage.py [hunk 12]
```diff
@@ -0,0 +1,399 @@
...
+        dispatch_mock.assert_not_awaited()
+
+    async def test_raw_recv_ignores_unexpected_message_type(self, websocket_service: WebsocketService) -> None:
+        """raw_recv logs and returns without dispatching or raising for an unhandled frame type."""
+        fake_ws = build_fake_ws()
+        fake_ws.receive = AsyncMock(return_value=SimpleNamespace(type=WSMsgType.PONG, data=None))
+        websocket_service._ws = fake_ws
+
+        dispatch_mock = AsyncMock()
+        websocket_service.dispatch = dispatch_mock  # collaborator of raw_recv
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/unit/core/test_websocket_service_coverage.py]: Fixed — same prefix applied.

File: tests/unit/core/test_websocket_service_coverage.py [hunk 13]
```diff
@@ -0,0 +1,399 @@
...
+    async def test_raw_recv_swallows_invalid_json_without_dispatching(
+        self, websocket_service: WebsocketService
+    ) -> None:
+        """raw_recv catches JSONDecodeError on malformed TEXT frames and skips dispatch."""
+        fake_ws = build_fake_ws()
+        fake_ws.receive = AsyncMock(return_value=SimpleNamespace(type=WSMsgType.TEXT, data="{not valid json"))
+        websocket_service._ws = fake_ws
+
+        dispatch_mock = AsyncMock()
+        websocket_service.dispatch = dispatch_mock  # collaborator of raw_recv
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/unit/core/test_websocket_service_coverage.py]: Fixed — same prefix applied.

File: tests/unit/core/test_websocket_service_coverage.py [hunk 14]
```diff
@@ -0,0 +1,399 @@
...
+            raise RuntimeError("boom")
+
+        websocket_service.dispatch_hass_event = failing_dispatch_hass_event  # collaborator of dispatch
+
+        await websocket_service.dispatch({"type": "event", "event": {}})  # must not raise
+
+    async def test_dispatch_ignores_unknown_message_type(self, websocket_service: WebsocketService) -> None:
+        """dispatch() falls through to the 'other' match case for a type it doesn't recognize."""
+        respond_mock = Mock()
+        websocket_service.respond_if_necessary = respond_mock  # collaborator of dispatch
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/unit/core/test_websocket_service_coverage.py]: Fixed — updated to `# boundary-exempt: collaborator of dispatch`.

File: tools/check_internal_patches.py [hunk 15]
```diff
@@ -231,19 +240,19 @@ def visit_Call(self, node: ast.Call) -> None:
...
 
-    Accepts the annotation anywhere on the statement's own physical lines, or on the
+    Accepts either annotation anywhere on the statement's own physical lines, or on the
     comment-only line immediately preceding it (1-based line numbers).
     """
     # lineno/end_lineno are 1-based; lines is 0-indexed, hence the -1 / -2 offsets.
     for i in range(lineno - 1, end_lineno):
-        if ANNOTATION in lines[i]:
+        if any(a in lines[i] for a in ANNOTATIONS):
             return True
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tools/check_internal_patches.py]: Fixed — `is_exempt()` now uses tokenize-derived comment tokens instead of raw substring search. Moved `extract_comments()` to `lint_helpers.py` as shared infrastructure. Added a regression test for the string-literal false-exemption case.
